lib/npm.js

```diff
@@ -295,6 +295,13 @@
           umask: umask
         }
 
+        if (config.get('proxy') && !process.env['HTTP_PROXY']){
+          process.env['HTTP_PROXY'] = config.get('proxy')
```

Comment:
There are no other cases where we run child processes, so if you do those two cases then you've covered everything.

Tests should be pretty easy regardless... set proxy config and check `npm run env` for changes made to `lib/utils/lifecycle.js`. Slightly trickier with `git`. My approach would be to provide a stub replacement for `git` that just prints out the value of `HTTP_PROXY`, then you can run something like `npm install git://…` and then look for the `HTTP_PROXY` message in the output.